app/Services/ProductService.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Services;
+
+use Illuminate\Support\Collection;
+
+class ProductService
+{
+    /**
+     * This function returns total price of given products
+     *
+     * @param Collection $products
+     * @return float
+     */
+    public function getTotalRevenue(Collection $products): float
+    {
+        $revenue =  $products->sum('price');
+        return roundNumber($revenue, 2);
+    }
+
+    /**
+     * This function returns total revenue in the first half of the month
+     *
+     * @param Collection $products
+     * @return float
+     */
+    public function getFirstHalfRevenue(Collection $products): float
```

Comment:
this is 100% fine.

As an alternative you could've used partition() to compute both firstHalfRevenue and secondHalfRevenue in one go.

[$firstHalfRevenue, $secondHalfRevenue] = $products
            ->partition(fn(array $product) => isFirstHalfDate($product->date_added))
            ->map(fn($halfProducts) => roundNumber($halfProducts->sum('price')));